Prepare DDL docs for GA (#14648)
docker/docker.github.io: desktop/linux/install/debian.md

---
description: Instructions for installing Docker Desktop on Debian
keywords: requirements, apt, installation, debian, install, uninstall, upgrade, update
title: Install Docker Desktop on Debian
toc_max: 4
---

To get started with Docker Engine on Debian, make sure you
[meet the prerequisites](#prerequisites), then
[install Docker](#install-docker-desktop).

## Prerequisites

{% include desktop-linux-requirements.md %}

> Additionally, the Gnome Desktop environment must have installed AppIndicator and KStatusNotifierItem [Gnome extensions](https://extensions.gnome.org/extension/615/appindicator-support/).

Reviewer: ```suggestion
Additionally, for a Gnome Desktop environment you must install AppIndicator and KStatusNotifierItem [Gnome extensions](https://extensions.gnome.org/extension/615/appindicator-support/).
```

### OS requirements

To install Docker Desktop, you need the 64-bit version of one of these Debian
versions:

- Debian 11

### Uninstall old versions

To remove Docker Desktop for Linux, run:

```console
$ sudo apt remove docker-desktop
```

For a complete cleanup, remove configuration and data files at `$HOME/.docker/desktop`, the symlink at `/usr/local/bin/com.docker.cli`, and purge
the remaining systemd service files.

```console
$ rm -r $HOME/.docker/desktop
$ sudo rm /usr/local/bin/com.docker.cli
$ sudo apt purge docker-desktop
```

> **Note**
>

> > Users that have installed Docker Desktop for Linux tech preview or beta version, need to remove all files that were generated by those packages(eg. `~/.config/systemd/user/docker-desktop.service`, `~/.local/share/systemd/user/docker-desktop.service`).

Reviewer: ```suggestion
> If you have installed the Docker Desktop for Linux tech preview or beta version, you need to remove all files that were generated by those packages(eg. `~/.config/systemd/user/docker-desktop.service`, `~/.local/share/systemd/user/docker-desktop.service`).
```

#### Install Docker Desktop

Recommended approach to install Docker Desktop on Debian:

1. [Set up Docker's deb repository](/engine/install/Debian.md##set-up-the-repository) 

2. Download latest deb package from the [release](../release-notes/index.md) page.

3. Install the package with apt as follows:
    
    ```console
    $ sudo apt-get update
    $ sudo apt-get install ./docker-desktop-<version>-<arch>.deb
    ```

> **Note**
>
> At the end of the installation process, `apt` displays an error due to installing a downloaded package. You

> > an ignore this error message.

Reviewer: ```suggestion
> can ignore this error message.
```

3. Install the package with apt as follows:
    
    ```console
    $ sudo apt-get update
    $ sudo apt-get install ./docker-desktop-<version>-<arch>.deb
    ```

> **Note**
>
> At the end of the installation process, `apt` displays an error due to installing a downloaded package. You
> can ignore this error message.
>
>  ```
>  N: Download is performed unsandboxed as root, as file '/home/user/Downloads/docker-desktop.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
>  ```

  There are a few post-install configuration steps done through the maintainers' scripts (post-install script contained  in the deb package.

  The post-install script:

  - sets the capability on the Docker Desktop binary to map privileged ports and set resource limits
  - adds a DNS name for Kubernetes to `/etc/hosts`
  - creates a link from `/usr/bin/docker` to `/usr/local/bin/com.docker.cli`

## Launch Docker Desktop

{% include desktop-linux-launch.md %}

#### Upgrade Docker Desktop

Once a new version for Docker Desktop is released, the Docker UI shows a notification. 
 You need to download the new package each time you want to upgrade Docker and run

>  ```console
>     $ sudo apt-get install ./docker-desktop-<version>-<arch>.deb
> ```

Reviewer 2: Some trailing whitespace

```suggestion
```console
$ sudo apt-get install ./docker-desktop-<version>-<arch>.deb
```
```
